### PR TITLE
made DNS resolver configurable

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: docs/configuration.txt [hunk 1]
```diff
@@ -511,6 +512,12 @@ Crate has built-in support for service discovery via DNS.
   The DNS query that is used to look up SRV records,
   usually in the format ``_service._protocol.fqdn``
 
+**discovery.srv.resolver**
+  | *Default:*  ``not set``
```
Comment (mfussenegger) [inline docs/configuration.txt]: isn't there a default ?

Comment (chaudum) [inline docs/configuration.txt]: default is `null` (using system resolver)

File: dns-discovery/src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java [hunk 2]
```diff
@@ -78,6 +93,7 @@ public SrvUnicastHostsProvider(Settings settings, TransportService transportServ
 
     protected Record[] lookupRecords() throws TextParseException {
         Lookup lookup = new Lookup(query, Type.SRV);
+        lookup.setResolver(this.resolver);
```
Comment (mfussenegger) [inline dns-discovery/src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java]: does setResolver work for null arguments?

**Comment by mfussenegger** [comment]
ack

**Pull request merged**